front: add path step row in new itinerary modal

SharglutDev opened the pull request:
close #13256 

> [!NOTE]
> To test the fields, add some empty path steps in `pathSteps` initial state 

> [!CAUTION]
> Fields are in readonly so it is normal their styles don't match exactly the mockup

SharglutDev committed:
ui-icons: add new itinerary modal icons

Signed-off-by: SharglutDev <p.filimon75@gmail.com>

SharglutDev committed:
front: add path step row in new itinerary modal

Signed-off-by: SharglutDev <p.filimon75@gmail.com>

Caracol3 commented on front/src/applications/operationalStudies/views/Scenario/components/ManageTimetableItem/Itinerary/PathStepItem.tsx:
```diff
@@ -0,0 +1,67 @@
+        readOnly
+      />
+      <div className="map-interactions">
+        <AddLocation size="lg" />
+        <FocusLocation size="lg" className={cx('focus-map-icon', { empty: !pathStep })} />
```
I know `aria-labels` aren’t consistently used in the app, but shouldn’t we at least add them for new buttons to improve accessibility?

SharglutDev replied:
Currently, I don't know how a blind / visually impaired user could add an op by clicking on the map but I totally agree that it is a good practice to have and we should add them every where it is necessary !

SharglutDev replied:
Done

Caracol3 commented:
In the mockup, the vertical line touches the round element directly, but in the app there’s a small gap above it. And below the round element, the line starts a bit too high.

Here in the app:
<img width="421" height="389" alt="image" src="https://github.com/user-attachments/assets/a4de2800-ba38-4ef0-bc29-79e37d34f5e7" />

In the mockup:
<img width="639" height="610" alt="image" src="https://github.com/user-attachments/assets/b5ec653a-1306-4da6-b3da-7c010a98726c" />

SharglutDev committed:
front: introduce new PathStep type

The type has less properties and will be often coupled with another one
containing the path step metadata.
Some unused properties (deleted, locked) which are not used in front end
yet have been removed.

Signed-off-by: SharglutDev <p.filimon75@gmail.com>

SharglutDev commented:
> In the mockup, the vertical line touches the round element directly, but in the app there’s a small gap above it. And below the round element, the line starts a bit too high.

Good catch ! Should be fixed.

Caracol3 commented on front/src/styles/scss/applications/operationalStudies/_itineraryModal.scss:
```diff
@@ -35,6 +41,117 @@
+          }
+        }
+
+        // this wrapper is be used to display the "adding location on map" style
```
Little typo here

SharglutDev replied:
Fixed :)

Caracol3 reviewed (approved):
LGTM (tested)

Math-R reviewed (approved):
lgtm
